torchci/lib/bot/retryBot.ts

```diff
@@ -1,7 +1,148 @@
 import { Probot } from "probot";
+import { getFlakyJobsFromPreviousWorkflow } from "../jobUtils";
 import { CachedConfigTracker } from "./utils";
 
+const SUCCESS_CONCLUSIONS = ["success"];
 const FAILURE_CONCLUSIONS = ["failure", "cancelled", "timed_out"];
+const ALLOWED_WORKFLOW_PREFIXES: { [key: string]: string[] } = {
+  pytorch: ["lint", "pull", "trunk", "linux-binary", "windows-binary"],
+  vision: [
+    "lint",
+    "Build Linux",
+    "Build Macos",
+    "Build M1",
+    "Tests on Linux",
+    "Tests on macOS",
+  ],
+};
+
+async function retryPreviousWorkflow(
+  ctx: any,
+  owner: string,
+  repo: string,
+  branch: string,
+  workflowName: string,
+  workflowId: number
+) {
+  const flakyJobs = await getFlakyJobsFromPreviousWorkflow(
+    owner,
+    repo,
+    branch,
+    workflowName,
+    workflowId
+  );
+
+  if (flakyJobs === undefined || flakyJobs.length === 0) {
+    return;
+  }
+
+  if (flakyJobs.length === 1) {
+    // If only one should be rerun, just rerun that job
+    return await ctx.octokit.rest.actions.reRunJobForWorkflowRun({
+      owner,
+      repo,
+      job_id: flakyJobs[0].job_id,
+    });
+  }
```

Comment:
We don't really need this bit. IIRC the original logic was designed to handle the case where you want to retry one job in the workflow but not the other

Reply:
Ah ok, make sense.  I'll clean this up.